project_euler/problem_148/sol1.py

```diff
@@ -0,0 +1,77 @@
+"""
+Project Euler Problem 148 : https://projecteuler.net/problem=148
+Author:	Sai Teja Manchi
+Problem Statement:
+We can easily verify that none of the entries in the
+first seven rows of Pascal's triangle are divisible by 7:
+                               1
+                          1          1
+                     1          2          1
+                1          3          3          1
+           1          4          6          4          1
+      1          5         10         10          5          1
+1          6         15         20         15          6          1
+However, if we check the first one hundred rows, we will find that
+only 2361 of the 5050 entries are not divisible by 7.
+Find the number of entries which are not divisible by 7
+in the first one billion (109) rows of Pascal's triangle.
+
+Solution:
+We iteratively generate each row in the pascal triangle one-by-one.
+Since Pascal's triangle is vertically-symmetric,
+We only need to generate half of the values.
+We then count the values which are not divisible by 7.
+We only store the remainders(when divided by 7) in the list to reduce memory usage.
+
+Note: In the original problem, we need to calculate for 10^9 rows
+      but we took 10^4 rows here by default.
+"""
+
+
+def solution(pascal_row_count: int = 10**4) -> int:
```

Comment:
try this:

`def solution(pascal_row_count: int = 10**9) -> int:`

Reply:
Hey,

I've tried with 10**9 but one of the tests failed due to timeout as it was taking more than 6 hours to run.

Reply:
It's not uncommon for brute force solutions to take a long time or run into timeout issues when dealing with very large inputs. In this case, the input value of 10**9 is extremely large, and the current implementation of the "solution" function does not have an efficient algorithm to handle such large inputs.